advanced-creation.md

```diff
@@ -111,3 +112,119 @@ const unicorn = got.create(settings);
 // Same as:
 const unicorn = got.extend({headers: {unicorn: 'rainbow'}});
 ```
+
+### Merging instances
+
```

Comment:
I think it would be nice to have something like this as an intro. Feel free to improve it.

> Got supports composing multiple instances together. This is very powerful. You can create a client that limits download speed and then compose it with an instance that signs a request. It's like plugins without any of the plugin mess. You just create instances and then compose them together.

Reply:
Sounds good!